contracts/RMRK/extension/soulbound/RMRKSoulbound.sol

```diff
@@ -25,28 +25,19 @@ abstract contract RMRKSoulbound is IERC6454alpha, RMRKCore {
         address to,
         uint256 tokenId
     ) internal virtual override {
-        if (
-            from != address(0) && // Exclude minting
-            to != address(0) && // Exclude Burning
-            !isTransferable(tokenId)
-        ) revert RMRKCannotTransferSoulbound();
+        if (!isTransferable(tokenId, from, to))
+            revert RMRKCannotTransferSoulbound();
 
         super._beforeTokenTransfer(from, to, tokenId);
     }
 
-    /**
-     * @inheritdoc IERC6454alpha
-     */
-    function isTransferable(uint256) public view virtual returns (bool) {
-        return false;
-    }
-
     function isTransferable(
-        uint256 tokenId,
-        address,
-        address
+        uint256,
+        address from,
+        address to
     ) public view virtual returns (bool) {
-        return isTransferable(tokenId);
+        return ((from == address(0) || // Exclude minting
+            to == address(0)) && from != to); // Exclude Burning // Besides the obvious transfer to self, if both are address 0 (general transferability check), it returns false
```

Comment:
I was initially worried that this clashes with the check for general transferability, but I was wrong 😂 
```solidity
from != to
```